### ISSUE TITLE
Scrypt docs code example contradicts RFC 7914

### ISSUE THREAD

**Issue by Badg** [pyca/cryptography#3302]
[Current docs for scrypt](https://cryptography.io/en/latest/hazmat/primitives/key-derivation-functions/#cryptography.hazmat.primitives.kdf.scrypt.Scrypt) give a code example:

```python
>>> kdf = Scrypt(
...     salt=salt,
...     length=64,
...     n=1024,
...     r=8,
...     p=16,
...     backend=backend
... )
```

**Note the value of ```p``` above.** This contradicts both the subsequent text:

> RFC 7914 recommends values of r=8 and **p=1** while scaling n to the number appropriate for your system.

as well as the RFC itself.

Additionally, though it's a much more minor thing, the code example has an unused ```from cryptography.hazmat.primitives import hashes``` which would trip code linters by default.

As a side note, I'm correct in saying the Windows wheels link against a version of openSSL that doesn't support scrypt, right? Would it be worthwhile to add that as a ```.. note::``` directive in the Scrypt section to decrease the likelihood of future issues getting raised?

I'm submitting a PR for this momentarily but can't actually test it locally, because I'm on the Windows wheel build (with openSSL 1.0.2j).

**Comment by alex** [comment]
You're correct that our current Windows wheels are still on OpenSSL 1.0.2. We expect to move them to 1.1.0 in the next release or two.

**Comment by Badg** [comment]
Hmm, thinking about this a little more. The [scrypt paper itself](https://www.tarsnap.com/scrypt/scrypt.pdf) (page 13, ctrl+F "interactive logins" recommends n = 2**14 for interactive logins and 2**20 for sensitive files. Clearly 1024 is insufficient here. I'll update the PR (#3303) accordingly.

**Comment by Badg** [comment]
Closed by #3303
